scripts/determine-merge-branches.ts

```diff
@@ -0,0 +1,66 @@
+import {join} from 'path';
+import * as semver from 'semver';
+import {exec} from 'shelljs';
+
+/** Regular expression that matches remote head refs. */
+const UPSTREAM_HEAD_REF = /refs\/heads\/(.*)$/;
+
+/** Determines merge branches based on the current project version. */
+export function determineMergeBranches(
+    owner: string, name: string): {minor: string, patch: string} {
+  const projectRoot = join(__dirname, '../');
+  const currentVersion = semver.parse(require('../package.json').version);
+
+  if (currentVersion === null) {
+    throw Error('Cannot parse version set in project "package.json" file.');
+  }
+
+  const {major, minor, patch, prerelease} = currentVersion;
+  const isMajor = minor === 0 && patch === 0;
+  const isMinor = minor !== 0 && patch === 0;
+
+  // If there is no prerelease, then we compute patch and minor branches based
+  // on the current version major and minor.
+  if (prerelease.length === 0) {
+    return {minor: `${major}.x`, patch: `${major}.${minor}.x`};
+  }
+
+  // If current version is set to a minor prerelease, we can compute the merge branches
+  // statically. e.g. if we are set to `9.3.0-next.0`, then our merge branches should
+  // be set to `9.x` and `9.2.x`.
+  if (isMinor) {
+    return {minor: `${major}.x`, patch: `${major}.${minor - 1}.x`};
+  } else if (!isMajor) {
+    throw Error('Unexpected version. Cannot have prerelease for patch version.');
+  }
+
+  // If we are set to a major prerelease, we cannot statically determine the stable
+  // patch branch (as the latest minor segment is unknown). We determine it by looking
+  // for existing patch branches upstream.
```

Comment:
Not in this PR, but one idea for later would be to check npm for this info

Reply:
yeah, that is a great idea. I based the current Git logic on what has been done in framework, but consulting NPM sounds like something we should consider too.

This sounds like it could be more reliable. Only small "issue" will be that we'd need to rely on a single package on NPM, while with Git we'd rely on the actual overall monorepo/repo state.